### PR TITLE
Use Google storage scope to show contents of files [BA-4992]

### PR DESCRIPTION
**Pull request by rsasch**
If user is using Google authentication and adds the `"https://www.googleapis.com/auth/devstorage.read_only"` scope to their `capabilities_config.json`, the UI will allow them to see the contents of logs (and gray out icons when the logs don't exist or are empty) up to 1 MB.

Screen Shot 2019-05-29 at 3 09 07 PM

When the user clicks on a log icon, they will see the contents in a dialog box:
Screen Shot 2019-05-29 at 3 09 20 PM

Closes https://broadworkbench.atlassian.net/browse/BA-4992

### CODE REVIEW

File: ui/src/app/core/gcs.service.ts [hunk 1]
```diff
@@ -7,29 +7,23 @@ declare const gapi: any;
...
+      if (authenticated && this.authService.gcsReadAccess) {
+        return Promise.resolve(true);
       }
-
-      return Promise.reject({
-        status: 401,
-        title: 'Unauthorized',
-        message: 'Authentication failed, please try signing in again.'
-      });
-    });
```
Comment (salonishah11) [inline ui/src/app/core/gcs.service.ts]: What http response gets returned to user now if they are unauthorized?

File: ui/src/app/job-details/common/debug-icons/debug-icons.component.ts [hunk 2]
```diff
@@ -1,36 +1,99 @@
...
+  }
+
+  private async getLogContents(url: string): Promise<string> {
+    const bucket = ResourceUtils.getResourceBucket(url);
+    const object = ResourceUtils.getResourceObject(url);
+    return await this.gcsService.readObject(bucket, object)
+      .then(data => data);
+  }
+
+  private getFileName(filePath: string): string {
```
Comment (mcovarr) [inline ui/src/app/job-details/common/debug-icons/debug-icons.component.ts]: TOL is there not a library function for this?

File: ui/src/app/core/gcs.service.ts (hunk 1 above)
Comment (rsasch) [reply ui/src/app/core/gcs.service.ts]: added in error handling

File: ui/src/app/job-details/common/debug-icons/debug-icons.component.html [hunk 3]
```diff
@@ -9,25 +9,25 @@
...
         <span>stderr log</span>
       </clr-tooltip-content>
-    </a>
+    </button>
   </clr-tooltip>
-  <clr-icon *ngIf="!getTaskDirectory(directory)" shape="folder" class="is-solid disabled"></clr-icon>
-  <clr-tooltip *ngIf="!!getTaskDirectory(directory)">
-    <a class="log-item" [href]="getTaskDirectory(directory)" target="_blank">
+  <clr-icon *ngIf="!getDirectoryUrl(directory)" shape="folder" class="is-solid disabled"></clr-icon>
+  <clr-tooltip *ngIf="!!getDirectoryUrl(directory)">
```
[the remark this answers is not in the document]
Comment (rsasch) [reply ui/src/app/job-details/common/debug-icons/debug-icons.component.html]: Yup, `!!` casts to boolean, but it doesn't look like it was necessary here, so removed.

File: ui/src/app/job-details/common/debug-icons/debug-icons.component.spec.ts [hunk 4]
```diff
@@ -57,22 +59,21 @@ describe('JobDebugIconsComponent', () => {
     expect(de.queryAll(By.css('clr-icon[shape=exclamation-triangle]')).length).toEqual(1);
   }));
 
-  it('should link to the right location for stdout', async(() => {
+  it('should calculate to the right location for stdout', async(() => {
```
[the remark this answers is not in the document]
Comment (rsasch) [reply ui/src/app/job-details/common/debug-icons/debug-icons.component.spec.ts]: fixed

File: ui/src/app/job-details/common/debug-icons/debug-icons.component.spec.ts [hunk 5]
```diff
@@ -57,22 +59,21 @@ describe('JobDebugIconsComponent', () => {
...
     fixture.detectChanges();
-    let de: DebugElement = fixture.debugElement;
-    expect(de.queryAll(By.css('a.log-item'))[0].nativeElement.href).toEqual('https://console.cloud.google.com/storage/browser/test-bucket/test-job?prefix=stdout.txt');
+    expect(testComponent.jobDebugIconsComponent.getResourceUrl(job.stdout) == 'https://console.cloud.google.com/storage/browser/test-bucket/test-job?prefix=stdout.txt');
   }));
 
-  it('should link to the right location for stderr', async(() => {
+  it('should calculate to the right location for stderr', async(() => {
```
[the remark this answers is not in the document]
Comment (rsasch) [reply ui/src/app/job-details/common/debug-icons/debug-icons.component.spec.ts]: fixed

File: ui/src/app/job-details/common/debug-icons/debug-icons.component.ts (hunk 2 above)
Comment (rsasch) [reply ui/src/app/job-details/common/debug-icons/debug-icons.component.ts]: not that I could find

**Pull request merged**